esp_driver_sccb/sccb_i2c/include/sccb_i2c.h

```diff
@@ -0,0 +1,39 @@
+/*
+ * SPDX-FileCopyrightText: 2024 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "driver/i2c_master.h"
+#include "sccb_types.h"
+
+/**
+ * @brief Type of sccb i2c controller handle
+ */
+typedef struct sccb_i2c_controller_t *esp_sccb_i2c_handle_t;
+
+/**
+ * @brief ESP SCCB I2C configuration
+ */
+typedef struct {
+    i2c_port_num_t i2c_port;              ///< I2C port number, `-1` for auto selecting
+    gpio_num_t sda_io_num;                ///< GPIO number of I2C SDA signal, pulled-up internally
+    gpio_num_t scl_io_num;                ///< GPIO number of I2C SCL signal, pulled-up internally
+    i2c_addr_bit_len_t dev_addr_length;   ///< Select the address length of the slave device
+    uint16_t device_address;              ///< I2C device raw address. (The 7/10 bit address without read/write bit)
+    uint32_t scl_speed_hz;                ///< I2C SCL line frequency
+    sccb_clock_source_t clk_source;       ///< Clock source of I2C master bus, channels in the same group must use the same clock source
+} esp_sccb_i2c_config_t;
+
+/**
+ * @brief new esp sccb i2c controller
+ *
+ * @param[in] config       esp sccb i2c configuration
+ * @param[in] ret_handle   esp sccb handle
+ *
+ * @return
+ *         - ESP_ERR_INVALID_ARG   If the combination of arguments is invalid.
+ *         - ESP_ERR_NO_MEM        If out of memory
+ *         - ESP_OK                On success
+ */
+esp_err_t esp_sccb_new_i2c_ctlr(const esp_sccb_i2c_config_t *config, esp_sccb_handle_t *ret_handle);
```

Comment:
question, if the user wants to use the same physical I2C bus for the camera and other i2c slave sensros, how can he do? Because now inside this function, we're allocating the I2C bus, this will take the ownership, and user can't get the same bus outside the sccb driver

Reply:
I will add a note in the header saying the I2C bus is owned by this API.

Reply:
how about let the user provide the I2C bus handle, and the sccb driver uses that bus handle?

Reply:
Yes, the classic usage scenario is:
1) Call `i2c_new_master_bus(&i2c_mst_config, &bus_handle)` to initialize the I2C bus, obtain the handler, and then use `bus_handle` to control the sub-device(some sensors, just like led & motor)
2) Then use the `bus_handle` to control the camera sensors on the sccb bus.  
Maintaining the reusability of this I2C port is an important issue.